tests/badsymbols.pl

```diff
@@ -89,6 +89,7 @@ sub scanenums {
                ($_ ne "typedef") &&
                ($_ ne "enum") &&
                ($_ !~ /^[ \t]*$/) &&
+               ($_ ne "#line") &&
```

Comment:
What compiler are you using? I seems like a better idea would be to make line 70 make the `#line` check match, as that's designed so that the script doesn't care about include files that aren't in curl's include dir, and this change you propose suggests that the check on line 70 didn't work for your preprocessor....

Reply:
Hm, or maybe it mismatches when it is still within the same file? How does this error appear for you?

Reply:
Hello.
I am using Fujitsu C/C++ compiler version 03.2E61, its output a bit different from GCC's (please see the example below).
```
# 1015 "./../include/curl/curl.h"	// GCC
#line 1015 "./../include/curl/curl.h"	// Fujitsu C/C++
```
Please take a closer look at line 70 and the code block below. The regular expression `/^#(line|) (\d+) \"(.*)\"/` matches both GCC and Fujitsu C/C++ output as expected. But lines like `# 1015 "./../include/curl/curl.h"` or `#line 1015 "./../include/curl/curl.h"` are processed further when the specified full name of a header file contains `$incdir`.

The code in lines 85-87 reduces that output to `#` and `#line` respectively. As I can see from the code, the case for GCC is handled in line 92 `($_ ne "#")`, so such lines are dropped. My solution is similar to the existing one for GCC, but not the only possible, that is for sure.

I can provide Fujitsu C/C++ preprocessor's output for that test, if needed.

Reply:
> I seems like a better idea would be to make line 70 make the #line check match, as that's designed so that the script doesn't care about include files that aren't in curl's include dir

There is another (and probably better) solution that works for GCC and Fujitsu C/C++, please see the attached patch.
[badsymbols.pl.patch.txt](https://github.com/curl/curl/files/4676728/badsymbols.pl.patch.txt)

Reply:
Here is an example of Fujitsu C/C++ preprocessor's output, that causes test 1167 fail.
```
typedef enum {

  CURLMOPT_SOCKETFUNCTION = 20000 + 1,
#line 352 "./../include/curl/multi.h"
CURLMOPT_SOCKETDATA = 10000 + 2,
#line 355 "./../include/curl/multi.h"
CURLMOPT_PIPELINING = 0 + 3,
#line 358 "./../include/curl/multi.h"
CURLMOPT_TIMERFUNCTION = 20000 + 4,
#line 361 "./../include/curl/multi.h"
CURLMOPT_TIMERDATA = 10000 + 5,
#line 364 "./../include/curl/multi.h"
CURLMOPT_MAXCONNECTS = 0 + 6,
#line 367 "./../include/curl/multi.h"
CURLMOPT_MAX_HOST_CONNECTIONS = 0 + 7,
#line 370 "./../include/curl/multi.h"
CURLMOPT_MAX_PIPELINE_LENGTH = 0 + 8,
#line 374 "./../include/curl/multi.h"
CURLMOPT_CONTENT_LENGTH_PENALTY_SIZE = 30000 + 9,
#line 378 "./../include/curl/multi.h"
CURLMOPT_CHUNK_LENGTH_PENALTY_SIZE = 30000 + 10,
#line 382 "./../include/curl/multi.h"
CURLMOPT_PIPELINING_SITE_BL = 10000 + 11,
#line 386 "./../include/curl/multi.h"
CURLMOPT_PIPELINING_SERVER_BL = 10000 + 12,
#line 389 "./../include/curl/multi.h"
CURLMOPT_MAX_TOTAL_CONNECTIONS = 0 + 13,
#line 392 "./../include/curl/multi.h"
CURLMOPT_PUSHFUNCTION = 20000 + 14,
#line 395 "./../include/curl/multi.h"
CURLMOPT_PUSHDATA = 10000 + 15,
#line 398 "./../include/curl/multi.h"
CURLMOPT_MAX_CONCURRENT_STREAMS = 0 + 16,

  CURLMOPT_LASTENTRY
} CURLMoption;
```

Reply:
Your example lines and your example snippet show `#line` lines that *should* match the existing regex. Right? 

I agree that your updated patch looks better. Can you use that here in PR instead, force-push it the branch?

Reply:
> Your example lines and your example snippet show #line lines that should match the existing regex. Right?
Yes.

> Can you use that here in PR instead, force-push it the branch?
Sure.